ee/clickhouse/queries/session_recordings/clickhouse_session_recording_list.py

```diff
@@ -1,119 +1,234 @@
 
 
 class EventFiltersSQL(NamedTuple):
-    event_select_clause: str
-    event_where_clause: str
     aggregate_select_clause: str
-    aggregate_where_clause: str
+    aggregate_having_clause: str
+    where_conditions: str
     params: Dict[str, Any]
 
 
 class ClickhouseSessionRecordingList(SessionRecordingList):
-    _recording_duration_select_statement = (
-        "dateDiff('second', toDateTime(MIN(timestamp)), toDateTime(MAX(timestamp))) as duration,"
-    )
-    _recording_full_snapshot_select_statement = "COUNT((JSONExtractInt(snapshot_data, 'type') = 2 OR JSONExtractBool(snapshot_data, 'has_full_snapshot')) ? 1 : NULL) as full_snapshots"
-    _session_recording_event_table = "session_recording_events"
-    _session_recording_select_statements = """
-            any(session_recordings.start_time) as start_time,
-            any(session_recordings.end_time) as end_time,
-            any(session_recordings.duration) as duration,
-            any(filtered_events.distinct_id) as distinct_id
-    """
-
-    _event_query = """
+    _session_recordings_query_with_entity_filter: str = """
+    SELECT
+        session_recordings.session_id,
+        any(session_recordings.start_time) as start_time,
+        any(session_recordings.end_time) as end_time,
+        any(session_recordings.duration) as duration,
+        any(session_recordings.distinct_id) as distinct_id,
+        arrayElement(groupArray(current_url), 1) as start_url,
+        arrayElement(groupArray(current_url), -1) as end_url
+        {event_filter_aggregate_select_clause}
+    FROM (
         SELECT
+            distinct_id,
+            event,
+            team_id,
             timestamp,
-            distinct_id
-            {event_filter_event_select_clause}
+            properties,
+            JSONExtractString(properties, '$current_url') as current_url
         FROM events
+        WHERE
+            team_id = %(team_id)s
+            {event_filter_where_conditions}
+            {events_timestamp_clause}
+    ) AS filtered_events
+    RIGHT OUTER JOIN (
+        SELECT
+            session_id,
+            MIN(timestamp) AS start_time,
+            MAX(timestamp) AS end_time,
+            dateDiff('second', toDateTime(MIN(timestamp)), toDateTime(MAX(timestamp))) as duration,
+            any(distinct_id) as distinct_id,
+            COUNT((JSONExtractInt(snapshot_data, 'type') = 2 OR JSONExtractBool(snapshot_data, 'has_full_snapshot')) ? 1 : NULL) as full_snapshots
+        FROM session_recording_events
         WHERE
             team_id = %(team_id)s
             {events_timestamp_clause}
-            {event_filter_event_where_clause}
+        GROUP BY session_id
+        HAVING full_snapshots > 0
+        {recording_start_time_clause}
+        {duration_clause} 
+    ) AS session_recordings
+    ON session_recordings.distinct_id = filtered_events.distinct_id
+    JOIN (
+        SELECT * FROM person_distinct_id WHERE person_distinct_id.team_id = %(team_id)s
```

Comment:
This subquery is wrong and will result in incorrect data since it doesn't account for how we mutate data in clickhouse.

See ee/clickhouse/queries/event_query.py for an example on how to join - use `GET_TEAM_PERSON_DISTINCT_IDS`